### DISCUSSION TITLE
Are connection pool beneficial for Trino JDBC?

### DISCUSSION THREAD

**Discussion by gaeljw** [trinodb/trino#13784] [Q&A] [answered]
Hi,

We are using Trino JDBC driver in a JVM application which also uses HikariCP for managing connections pool.

We noticed issues with our application pooling and are about to investigate which setting to tune but I'm wondering if this even make sense at all to use connection pools with Trino?
As far as I know, the JDBC driver is a actually a wrapper over HTTP connections, right?

For the full context, we are using Trino with long running (in the minute order of magnitude) streaming queries, thus I expect we have some tuning to do anyway.

Thanks for any feedback on this topic,

**Comment by findepi** [comment, marked as answer]
Trino JDBC Connection is lightweight, no point in pooling these.

**Comment by findepi** [comment]
However, a potential side-effect / benefit of using a connection pool can be application throttling. A pool of size N will prevent (typical) application from running more N queries at a time. 
(Ignoring the fact that a single Connection object can be used to run multiple queries at a time)

**Reply by gaeljw** [reply]
Sure, we might still setup a pool but more to accommodate the target usage of our application (and what Trino will allow) rather than to optimize performance of the connections themselves and context/switch.
Which is usually the opposite way of defining a pool (roughly  number of cores + some extra).
Thanks for the answer ;)

**Reply by findepi** [reply]
BTW https://trino.io/docs/current/admin/resource-groups.html is the recommended way of managing # concurrent queries (per cluster, per application/source, etc.)

**Reply by gaeljw** [reply]
Ah right, thanks for remembering this. I had seen it but forgot about it :)
